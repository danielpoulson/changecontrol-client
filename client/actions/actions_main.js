import axios from 'axios';
import config from '../../configEnv';

export const SET_CHANGE_STATE = 'SET_CHANGE_STATE';
export const SET_MAIN = 'SET_MAIN';
export const SET_USER = 'SET_USER';
export const USER_LOGGED_OUT = 'USER_LOGGED_OUT';
export const SET_FILETAB_COUNT = 'SET_FILETAB_COUNT';
export const SET_LOADING = 'SET_LOADING';
export const SET_USER_DASHBOARD = 'SET_USER_DASHBOARD';
export const SET_USER_FROM_SESSION_STATE = 'SET_USER_FROM_SESSION_STATE';

export function addUserDashboard(dashboard: any) {
  return { type: SET_USER_DASHBOARD, payload: dashboard };
}

export function getUserDashboard(username) {
  return (dispatch: Function) => {
    axios
      .get(`${config.serverURL}/api/changes/userdashboard/${username}`)
      .then(response => {
        dispatch(addUserDashboard(response.data));
      })
      .catch(error => {
        console.error('axios error', error); // eslint-disable-line no-console
      });
  };
}

export function setChangeState() {
  return { type: SET_CHANGE_STATE };
}

export function setMain(data) {
  return { type: SET_MAIN, data };
}

export function setFiletabCount(data) {
  return { type: SET_FILETAB_COUNT, data };
}

export function setLoading(data) {
  return { type: SET_LOADING, data };
}

export function setTitle(data) {
  return { type: 'SET_TITLE', data };
}

export function setUserFromSessionState() {
  return { type: SET_USER_FROM_SESSION_STATE };
}

export function setReturnedUser(request: Object) {
  return { type: SET_USER, payload: request };
}

export function setUser() {
  return (dispatch: Function) => {
    axios
      .get(`${config.serverURL}/api/users/logged`)
      .then(response => {
        dispatch(setReturnedUser(response.data));
      })
      .catch(error => {
        console.error('axios error', error); // eslint-disable-line no-console
      });
  };
}

export function addLogin(user: any) {
  return { type: SET_USER, payload: user };
}

export function login(username) {
  const url = `${config.serverURL}/login`;
  return (dispatch: Function) => {
    axios
      .post(url, username)
      .then(response => {
        dispatch(addLogin(response.data));
      })
      .catch(error => {
        console.error('axios error', error); // eslint-disable-line no-console
      });
  };
}

export function resetUser() {
  return { type: USER_LOGGED_OUT };
}

export function logoutUser() {
  return (dispatch: Function) => {
    axios
      .get(`${config.serverURL}/logout`)
      .then(() => {
        dispatch(resetUser());
      })
      .catch(error => {
        console.error('axios error', error); // eslint-disable-line no-console
      });
  };
}
