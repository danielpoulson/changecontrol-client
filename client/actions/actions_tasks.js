//SYNC VER.002 DP
import axios from 'axios';
import { baseURL } from '../utils/helpers';

export const GET_TASKS = 'GET_TASKS';
export const GET_TASK = 'GET_TASK';
export const ADD_TASK = 'ADD_TASK';
export const EDIT_TASK = 'EDIT_TASK';
export const DELETE_TASK = 'DELETE_TASK';
export const LOAD_PAGE_TASKS = 'LOAD_PAGE_TASKS';
export const GET_PROJECT_TASKS = 'GET_PROJECT_TASKS';
export const GET_ALL_TASKS = 'GET_ALL_TASKS';
export const SET_CAPA = 'SET_CAPA';

export function getTasks(data) {
  const url = `${baseURL}/api/tasks/${data}`;
  const request = axios.get(url).catch(error => {
    console.error('axios error', error); // eslint-disable-line no-console
  });

  return {
    type: GET_TASKS,
    payload: request
  };
}

export function getAllTasks() {
  const _status = 4;
  const _capa = 0;
  const url = `${baseURL}/api/tasks/all/${_status}/${_capa}`;
  const request = axios.get(url).catch(error => {
    console.error('axios error', error); // eslint-disable-line no-console
  });

  return {
    type: GET_TASKS,
    payload: request
  };
}

export function getProjectTasks(data) {
  const url = `${baseURL}/api/tasks/project/${data}`;
  const request = axios.get(url).catch(error => {
    console.error('axios error', error); // eslint-disable-line no-console
  });

  return {
    type: GET_PROJECT_TASKS,
    payload: request
  };
}

export function getTask(data) {
  let request = {};

  if (data !== 'new') {
    const url = `${baseURL}/api/tasks/${data}`;
    request = axios.get(url).catch(error => {
      console.error('axios error', error); // eslint-disable-line no-console
    });
  }

  return {
    type: GET_TASK,
    payload: request
  };
}

export function addTask(data) {
  const url = `${baseURL}/api/tasks`;
  const request = axios.post(url, data).catch(error => {
    console.error('axios error', error); // eslint-disable-line no-console
  });

  return {
    type: ADD_TASK,
    payload: request
  };
}

export function editTask(data) {
  const url = `${baseURL}/api/tasks/${data._id}`;
  axios.put(url, data).catch(error => {
    console.error('axios error', error); // eslint-disable-line no-console
  });

  return {
    type: EDIT_TASK,
    payload: data
  };
}

export function deleteTask(data) {
  const url = `${baseURL}/api/tasks/${data}`;
  axios.delete(url).catch(error => {
    console.error('axios error', error); // eslint-disable-line no-console
  });

  return {
    type: DELETE_TASK,
    payload: data
  };
}

export function loadPageTask(data) {
  return {
    type: LOAD_PAGE_TASKS,
    data
  };
}

export function exportTasks(search) {
  const url = `${baseURL}/api/tasks/export`;
  const request = axios.post(url, search).catch(error => {
    console.error('axios error', error); // eslint-disable-line no-console
  });

  return {
    type: 'ADD_EXPORTFILE',
    payload: request
  };
}

export function setCapa() {
  return {
    type: SET_CAPA
  };
}
