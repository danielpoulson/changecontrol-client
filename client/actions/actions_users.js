//SYNC 11/03/2017 DP
import axios from 'axios';
import { serverURL } from '../utils/helpers';

export const GET_USER = 'GET_USER';
export const GET_USERS = 'GET_USERS';
export const RESET_USER = 'RESET_USER';
export const SAVE_USER = 'SAVE_USER';
export const DELETED_USER = 'DELETED_USER';
export const USER_CREATED = 'USER_CREATED';

export function setUsers(users) {
  return { type: GET_USERS, payload: users };
}

export function getUsers() {
  return (dispatch: Function) => {
    axios
      .get(`${serverURL}/api/users/all`)
      .then(response => {
        dispatch(setUsers(response.data));
      })
      .catch(error => {
        console.error('axios error', error); // eslint-disable-line no-console
      });
  };
}

export function addUser(user) {
  return { type: GET_USER, payload: user };
}

export function getUser(id) {
  return (dispatch: Function) => {
    axios
      .get(`${serverURL}/api/users/${id}`)
      .then(response => {
        dispatch(addUser(response.data));
      })
      .catch(error => {
        console.error('axios error', error); // eslint-disable-line no-console
      });
  };
}

export function resetUser() {
  return {
    type: RESET_USER
  };
}

export function addNewUser(fullname) {
  return {
    type: USER_CREATED,
    fullname
  };
}

export function createUser(data) {
  return (dispatch: Function) => {
    axios
      .post(`${serverURL}/api/users`, data)
      .then(() => {
        dispatch(addNewUser(data.fullname));
      })
      .catch(error => {
        console.error('axios error', error); // eslint-disable-line no-console
      });
  };
}

export function saveUser(data) {
  const url = `${serverURL}/api/users/updateuser/${data.username}`;
  axios.put(url, data);

  return {
    type: SAVE_USER,
    data
  };
}

export function savePass(id, password) {
  const url = `${serverURL}/api/users/updatepass/${id}`;
  axios.put(url, { password });

  return {
    type: 'PASSWORD_CHANGED'
  };
}

export function removeUser(fullname) {
  return {
    type: DELETED_USER,
    fullname
  };
}

export function deleteUser(id, fullname) {
  return (dispatch: Function) => {
    axios
      .delete(`${serverURL}/api/users/${id}`)
      .then(() => {
        dispatch(removeUser(fullname));
      })
      .catch(error => {
        console.error('axios error', error); // eslint-disable-line no-console
      });
  };
}
