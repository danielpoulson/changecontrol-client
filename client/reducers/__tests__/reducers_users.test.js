import reducer from '../reducer_users';
import { addNewUser, removeUser, setUsers } from '../../actions/actions_users';

const UserList = ['Daniel Ricciardo', 'Sebastian Vettel', 'Lewis Hamilton', 'Valtteri Bottas'];

test('USER_CREATED should add a new user to the userlist', () => {
  const initialState = UserList;
  const newUser = ['Michael Schumacher'];
  const action = addNewUser(newUser);
  const newState = reducer(initialState, action);
  expect(newState).toMatchSnapshot();
});

test('GET_USERS should return a list of current users', () => {
  const initialState = [];

  const action = setUsers(UserList);
  const newState = reducer(initialState, action);
  expect(newState).toMatchSnapshot();
});

test('DELETE_USER should remove an item from the current users', () => {
  const initialState = [];
  const newUser = ['Lewis Hamilton'];
  const action = removeUser(newUser);
  const newState = reducer(initialState, action);
  expect(newState).toMatchSnapshot();
});
