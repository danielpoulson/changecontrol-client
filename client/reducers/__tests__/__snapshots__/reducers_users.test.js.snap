// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DELETE_USER should remove an item from the current users 1`] = `Array []`;

exports[`GET_USERS should return a list of current users 1`] = `
Array [
  "Daniel Ricciardo",
  "Sebastian Vettel",
  "Lewis Hamilton",
  "Valtteri Bottas",
]
`;

exports[`USER_CREATED should add a new user to the userlist 1`] = `
Array [
  "Daniel Ricciardo",
  "Sebastian Vettel",
  "Lewis Hamilton",
  "Valtteri Bottas",
  "Michael Schumacher",
]
`;
